arm-timeseriesinsights/2017-02-28-preview/Swagger/timeseriesinsights.json

```diff
@@ -0,0 +1,1780 @@
+        ],
+        "responses": {
+          "200": {
+            "description": "The reference data set definition was successfully updated and is in the response.",
+            "schema": {
+              "$ref": "#/definitions/ReferenceDataSetResource"
+            }
+          },
+          "404": {
+            "description": "The subscription, resource group, environment, or reference data set could not be found."
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "ReferenceDataSets"
+        ],
+        "operationId": "ReferenceDataSets_Delete",
+        "x-ms-examples": {
+          "ReferenceDataSetsDelete": { "$ref": "../examples/ReferenceDataSetsDelete.json" }
+        },
+        "description": "Deletes the reference data set with the specified name in the specified subscription, resource group, and environment",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/EnvironmentNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ReferenceDataSetNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The reference data set was successfully deleted."
+          },
+          "204": {
+            "description": "The reference data set was successfully deleted."
+          },
+          "404": {
+            "description": "The subscription, resource group, environment, or reference data set could not be found."
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.TimeSeriesInsights/environments/{environmentName}/referenceDataSets": {
+      "get": {
+        "tags": [
+          "ReferenceDataSets"
+        ],
+        "operationId": "ReferenceDataSets_ListByEnvironment",
+        "x-ms-examples": {
+          "ReferenceDataSetsListByEnvironment": { "$ref": "../examples/ReferenceDataSetsListByEnvironment.json" }
+        },
+        "description": "Lists all the available reference data sets associated with the subscription and within the specified resource group and environment.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/EnvironmentNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Reference data sets returned successfully.",
+            "schema": {
+              "$ref": "#/definitions/ReferenceDataSetListResponse"
```

Comment:
And we only allow 2 reference data sets per environment, so it will probably never need to be pageable.